src/sentry/api/exceptions.py

```diff
@@ -28,6 +24,17 @@ def __init__(self, code=None, message=None, detail=None, **kwargs):
         super(SentryAPIException, self).__init__(detail=detail)
 
 
+class ResourceMoved(SentryAPIException):
+    status_code = status.HTTP_302_FOUND
+    code = 'project-moved'
+    message = 'Project slug was renamed'
```

Comment:
Don't know if we should couple this with Projects, specifically. Can we make it more generic to cover other redirects in the future?